### PR TITLE
Support for new dataTime format in `DateTimeGranularitySpec` without explicitly setting size

### PR DESCRIPTION
**Pull request by s0nskar**
This is a follow-up PR for issue [#11028](https://github.com/apache/pinot/issues/11028)

Adding support for new dataTime format in `DateTimeGranularitySpec` without explicitly setting the size 
( ref comment by @Jackie-Jiang : https://github.com/apache/pinot/pull/11030#pullrequestreview-1515740305) 

Test Plan:
- Current UT passed
- Added new UT's

### CODE REVIEW

**Comment by s0nskar** [comment]
cc: @Jackie-Jiang @snleee PTAL

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeGranularitySpec.java [hunk 1]
```diff
@@ -53,38 +55,49 @@ public DateTimeGranularitySpec(String granularity) {
     char separator;
     int sizePosition;
     int timeUnitPosition;
+    int minTokens = NUM_TOKENS;
+    int maxTokens = NUM_TOKENS;
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeGranularitySpec.java:59]: (minor) Move the assignment into the if check for readability

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeGranularitySpec.java [hunk 2]
```diff
@@ -53,38 +55,49 @@ public DateTimeGranularitySpec(String granularity) {
...
+      _timeUnit = TimeUnit.valueOf(granularityTokens[timeUnitPosition]);
     } catch (Exception e) {
       throw new IllegalArgumentException(
-          String.format("Invalid size: %s in granularity: %s", granularityTokens[sizePosition], granularity));
+          String.format("Invalid time unit: %s in granularity: %s", granularityTokens[timeUnitPosition],
+              granularity));
+    }
+
+    // New format without explicitly setting size - use default size = 1
+    if (granularityTokens.length == 1) {
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeGranularitySpec.java:89]: (minor)
```suggestion
    if (granularityTokens.length == PIPE_FORMAT_MIN_TOKENS) {
```

**Review by Jackie-Jiang — APPROVED** [review]
LGTM with minor comments. Please take a look at the failed test

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeGranularitySpec.java (hunk 1 above)
Comment (s0nskar) [reply pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeGranularitySpec.java]: Done

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeGranularitySpec.java (hunk 2 above)
Comment (s0nskar) [reply pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeGranularitySpec.java]: Done

File: pinot-core/src/test/java/org/apache/pinot/core/operator/transform/function/DateTimeConversionTransformFunctionTest.java [hunk 3]
```diff
@@ -66,7 +66,7 @@ public Object[][] testIllegalArguments() {
         }, new Object[]{"dateTimeConvert(5,'1:MILLISECONDS:EPOCH','1:MINUTES:EPOCH','1:MINUTES')"}, new Object[]{
         String.format("dateTimeConvert(%s,'1:MILLISECONDS:EPOCH','1:MINUTES:EPOCH','1:MINUTES')", INT_MV_COLUMN)
     }, new Object[]{
-        String.format("dateTimeConvert(%s,'1:MILLISECONDS:EPOCH','1:MINUTES:EPOCH','MINUTES')", TIME_COLUMN)
+        String.format("dateTimeConvert(%s,'1:MILLISECONDS:EPOCH','1:MINUTES:EPOCH','MINUTES:1')", TIME_COLUMN)
```
Comment (s0nskar) [inline pinot-core/src/test/java/org/apache/pinot/core/operator/transform/function/DateTimeConversionTransformFunctionTest.java:69]: Fixing failing test, as `MINUTES` was getting considered as a valid granularity after this diff.

**Comment by s0nskar** [comment]
Thanks, @Jackie-Jiang for the review. Addressed comments.

**Comment by Jackie-Jiang** [comment]
Great job! Can you also help update the [Pinot doc](https://docs.pinot.apache.org/configuration-reference/schema#datetimefieldspec) reflecting this new format?

**Comment by s0nskar** [comment]
@Jackie-Jiang Will do 👍

**Pull request merged**